libs/math/include/math/utilities/ReadCSV.hpp

```diff
@@ -94,7 +94,21 @@ TensorType ReadCSV(std::string const &filename, math::SizeType const cols_to_ski
     }
     while (std::getline(ss, field_value, delimiter))
     {
-      weights(col, row) = fetch::math::Type<DataType>(field_value);
+      if (field_value.empty())
+      {
+        throw std::runtime_error("Empty field in ReadCSV");
+      }
+      if (unsafe_parsing)
+      {
+        // Constructing a fixed point from a double is not guaranteed to give the same results on
+        // different architectures and so is unsafe. But the fixed point string parsing does not
+        // support scientific notation.
```

Comment:
Maybe we should raise ticket for it. Or can't you just use it anyway?

Reply:
I've added the ticket